py/desispec/io/frame.py

```diff
@@ -169,3 +168,32 @@ def read_frame(filename, nspec=None):
         log.error("Frame did not pass simple vetting test. diagnosis={:d}".format(diagnosis))
     # Return
     return frame
+
+
+def search_for_framefile(frame_file):
+    """ Search for an input frame_file in the desispec redux hierarchy
+    Args:
+        frame_file:  str
+
+    Returns:
+        mfile: str,  full path to frame_file if found else raise error
+
+    """
+    log=get_logger()
+    # Parse frame file name
+    ifile = frame_file.split('/')[-1]
+    splits = ifile.split('-')
+    root = splits[0]
+    camera = splits[1]
+    fexposure = int(splits[2].split('.')[0])
```

Comment:
minor: `os.path.split()` and `os.path.splitext()` are more standard ways of doing some of this logic.